core/model/src/main/java/org/eclipse/rdf4j/model/IRI.java

```diff
@@ -72,4 +72,24 @@
 	 */
 	@Override
 	public int hashCode();
+
+	@Override
+	default boolean isBnode() {
+		return false;
+	}
+
+	@Override
+	default boolean isIRI() {
+		return true;
+	}
+
+	@Override
+	default boolean isLiteral() {
+		return false;
+	}
+
+	@Override
+	default boolean isTriple() {
+		return false;
+	}
```

Comment:
I suggest we move these defaults to `Value` instead, and then have each sub-interface override only the one for which the result should be true.

Reply:
I prefer to have Value have functional defaults (eg. instanceof), but it wouldn't actually make a difference I think.